neuralcompression/layers/_project_latent.py

```diff
@@ -0,0 +1,49 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Optional
+
+import torch.nn as nn
+
+from neuralcompression import VqVaeAutoencoderOutput
+
+
+class ProjectLatent(nn.Module):
+    """
+    Applies a convolution projection of the latent.
+
+    This class is intended to wrap a recursive tree of operations. The
+    projection is applied to the input, followed by whatever operation is
+    contained in ``child``. The output of ``child`` then has an inverse
+    projection applied.
+
+    Args:
+        input_dim: The input dimension for the projection.
+        output_dim: The output dimension for the projection.
+        child: A child operation to apply between forward and inverse
+            projections.
+    """
+
+    def __init__(
+        self, input_dim: int, output_dim: int, child: Optional[nn.Module] = None
+    ):
+        super().__init__()
+        self.child = child
```

Comment:
Same comment regarding the linked list in disguise, in that case it is not a linked list, but another data structure in disguise - but looking at the forward function, an nn.Sequential built by a factory function could do the job ?